_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+   loginRedirectUri="http://localhost:3000/callback" signup="false" %}
+
+Copy the output from the CLI and keep it for later.
+
+## Setup your Supabase project
+
+Go to your [Supabase account](https://app.supabase.com/) and click on **New Project**. Select your default organization (or create a new one). Name your project `OktaSupabase` and give it a strong password. Select the region closest to you and select the pricing plan that suits your needs best. Most likely this will be the `Free` plan but if you find a need to update to pro you can do that as well. Once you're done click **Create Project**.
+
+{% img blog/supabase-vue/supabase-create-project.png alt:"Create a project in Supabase" width:"600" %}{: .center-image }
+
+Make note of your `Project API Keys` and `Project Configuration` in the `Connecting to your new project` section of the project page. Specifically, you will need the `anon` `public` API key and the `URL`.
+
+### Create your table in Supabase
+
+Now add a new table to your database. In Supabase there are two ways to do this. You can use the SQL editor or you can use the GUI Table editor. I opted to use the Table editor to see how powerful the UI was and how easy it was to use. Once you open the table editor you can click **Create a new table**. Give the table the name `Entries` and a description if you want.
+
+Note that the `id` and `created_at` columns are prepopulated for you so you can skip these. Add the following columns to the table.
+
+> description: text
+> isPublic: bool
+> date: timestamptz
+> title: text
+
+Make each column non-nullable.  By default Supabase, makes all columns nullable. Then click **Save**. After a moment you will see your table in the _Table editor_ section.
+
+{% img blog/supabase-vue/supabase-create-table.png alt:"Create a table in supabase" width:"600" %}{: .center-image }
+
+### Populate some sample data
+
+Click on the table you just created in the table editor and click **Insert Row**. This will bring up an editor where you can add new data. I added 3 entries. Two public and one private entry. Give your entries dates that are in your current month. For example, I'm writing this article in September 2022 so I used three dates in that month.
+
+{% img blog/supabase-vue/supabase-add-row.png alt:"Add data to supabase" width:"600" %}{: .center-image }
+
+After each row click save and you should see the data you complete populated in the Table editor.
+
+## Create your Vue Application
+
+It's now time to create your Vue application. Start by running the following command.
+
+```sh
+npm init vue@3.3.2 OktaSupabase
+```
+
+This will create a new Vue application called `OktaSupabase` in a subfolder on your working directory with the same name. There will be several options in this command such as the project name, typescript support, JSX support, and others. See the image below for the options I've chosen. If you want to use some of these options feel free to turn them on.
```

Comment:
```suggestion
This will create a new Vue application called `OktaSupabase` in a subfolder on your working directory with the same name. There will be several options in this command such as the project name, TypeScript support, JSX support, and others. See the image below for the options I've chosen. If you want to use some of these options feel free to turn them on.
```